packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -196,34 +208,32 @@ const TreeNode = React.forwardRef<HTMLLIElement, TreeNodeProps>(
         event.stopPropagation();
       }
       if (match(event, keys.ArrowLeft)) {
-        const findParentTreeNode = (node) => {
+        const findParentTreeNode = (node: Element | null): Element | null => {
+          if (!node) return null;
           if (node.classList.contains(`${prefix}--tree-parent-node`)) {
             return node;
           }
           if (node.classList.contains(`${prefix}--tree`)) {
             return null;
           }
-          return findParentTreeNode(node.parentNode);
+          return findParentTreeNode(node.parentElement);
         };
         if (children && expanded) {
           if (!enableTreeviewControllable) {
             onToggle?.(event, { id, isExpanded: false, label, value });
           }
           setExpanded(false);
         } else {
-          /**
-           * When focus is on a leaf node or a closed parent node, move focus to
-           * its parent node (unless its depth is level 1)
-           */
-          findParentTreeNode(currentNode.current?.parentNode)?.focus();
+          const parentNode = findParentTreeNode(
+            currentNode.current?.parentElement || null
+          );
+          if (parentNode instanceof HTMLElement) {
+            parentNode.focus();
+          }
         }
       }
       if (children && match(event, keys.ArrowRight)) {
         if (expanded) {
-          /**
-           * When focus is on an expanded parent node, move focus to the first
-           * child node
-           */
```

Comment:
same as above, I think this comment is still relevant right?

Reply:
The comments are relevant, got missed in the cleanup 😅